Cannot make libdlk_x86_avx.a

Related #755 
I tried to run `make ar_x86_avx` but it failed for all models ([buildkite result](https://buildkite.com/blueoil/blueoil-test/builds/1516#_))

This is sample error log.
```
# make ar_x86_avx
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv5_conv2d_kernel_1_QTZ_binary_mean_scaling_new.cpp -o src/inputs/conv5_conv2d_kernel_1_QTZ_binary_mean_scaling_new.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv1_Const_1.cpp -o src/inputs/conv1_Const_1.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/pool7_reshape_shape.cpp -o src/inputs/pool7_reshape_shape.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv3_conv2d_kernel_1_QTZ_binary_mean_scaling_new.cpp -o src/inputs/conv3_conv2d_kernel_1_QTZ_binary_mean_scaling_new.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv7_kernel.cpp -o src/inputs/conv7_kernel.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv1_BatchNorm_FusedBatchNormV3_optimized_bias.cpp -o src/inputs/conv1_BatchNorm_FusedBatchNormV3_optimized_bias.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv6_BatchNorm_FusedBatchNormV3_optimized_bias.cpp -o src/inputs/conv6_BatchNorm_FusedBatchNormV3_optimized_bias.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv6_conv2d_kernel_1_QTZ_binary_mean_scaling_new.cpp -o src/inputs/conv6_conv2d_kernel_1_QTZ_binary_mean_scaling_new.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv6_BatchNorm_FusedBatchNormV3_optimized_scale.cpp -o src/inputs/conv6_BatchNorm_FusedBatchNormV3_optimized_scale.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv1_Const.cpp -o src/inputs/conv1_Const.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv1_conv2d_kernel_1_QTZ_binary_mean_scaling_new.cpp -o src/inputs/conv1_conv2d_kernel_1_QTZ_binary_mean_scaling_new.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv1_BatchNorm_FusedBatchNormV3_optimized_scale.cpp -o src/inputs/conv1_BatchNorm_FusedBatchNormV3_optimized_scale.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv4_conv2d_kernel_1_QTZ_binary_mean_scaling_new.cpp -o src/inputs/conv4_conv2d_kernel_1_QTZ_binary_mean_scaling_new.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv2_conv2d_kernel_1_QTZ_binary_mean_scaling_new.cpp -o src/inputs/conv2_conv2d_kernel_1_QTZ_binary_mean_scaling_new.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/inputs/conv7_bias.cpp -o src/inputs/conv7_bias.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/scaling_factors.cpp -o src/scaling_factors.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/thresholds.cpp -o src/thresholds.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/func/average_pool.cpp -o src/func/average_pool.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/func/conv2d.cpp -o src/func/conv2d.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/func/max_pool.cpp -o src/func/max_pool.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/func/pad.cpp -o src/func/pad.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/func/matmul.cpp -o src/func/matmul.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/func/softmax.cpp -o src/func/softmax.o 
g++ -I./include -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp -c src/func/lookup.cpp -o src/func/lookup.o 
src/func/lookup.cpp: In function ‘void func_Lookup(const TensorView<float, (MemoryLayout)7>&, const TensorView<QuantizedPacked<unsigned int>, (MemoryLayout)5>&, const TensorView<QuantizedPacked<unsigned int>, (MemoryLayout)5>&, const TensorView<QuantizedPacked<unsigned int>, (MemoryLayout)14>&)’:
src/func/lookup.cpp:43:43: warning: AVX vector return without AVX enabled changes the ABI [-Wpsabi]
   const auto coeff = _mm256_set1_ps(255.0f);
                                           ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:1285:1: error: inlining failed in call to always_inline ‘__m256 _mm256_set1_ps(float)’: target specific option mismatch
 _mm256_set1_ps (float __A)
 ^
src/func/lookup.cpp:43:43: error: called from here
   const auto coeff = _mm256_set1_ps(255.0f);
                                           ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:1451:1: error: inlining failed in call to always_inline ‘__m256 _mm256_castps128_ps256(__m128)’: target specific option mismatch
 _mm256_castps128_ps256 (__m128 __A)
 ^
src/func/lookup.cpp:45:78: error: called from here
     const auto vl0 = _mm256_castps128_ps256(_mm_loadu_ps(in_ptr + 3 * i +  0));
                                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:1451:1: error: inlining failed in call to always_inline ‘__m256 _mm256_castps128_ps256(__m128)’: target specific option mismatch
 _mm256_castps128_ps256 (__m128 __A)
 ^
src/func/lookup.cpp:46:78: error: called from here
     const auto vl1 = _mm256_castps128_ps256(_mm_loadu_ps(in_ptr + 3 * i +  4));
                                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:1451:1: error: inlining failed in call to always_inline ‘__m256 _mm256_castps128_ps256(__m128)’: target specific option mismatch
 _mm256_castps128_ps256 (__m128 __A)
 ^
src/func/lookup.cpp:47:78: error: called from here
     const auto vl2 = _mm256_castps128_ps256(_mm_loadu_ps(in_ptr + 3 * i +  8));
                                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:734:1: error: inlining failed in call to always_inline ‘__m256 _mm256_insertf128_ps(__m256, __m128, int)’: target specific option mismatch
 _mm256_insertf128_ps (__m256 __X, __m128 __Y, const int __O)
 ^
src/func/lookup.cpp:51:53: error: called from here
     const auto v0 = _mm256_insertf128_ps(vl0, vh0, 1);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:734:1: error: inlining failed in call to always_inline ‘__m256 _mm256_insertf128_ps(__m256, __m128, int)’: target specific option mismatch
 _mm256_insertf128_ps (__m256 __X, __m128 __Y, const int __O)
 ^
src/func/lookup.cpp:52:53: error: called from here
     const auto v1 = _mm256_insertf128_ps(vl1, vh1, 1);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:734:1: error: inlining failed in call to always_inline ‘__m256 _mm256_insertf128_ps(__m256, __m128, int)’: target specific option mismatch
 _mm256_insertf128_ps (__m256 __X, __m128 __Y, const int __O)
 ^
src/func/lookup.cpp:53:53: error: called from here
     const auto v2 = _mm256_insertf128_ps(vl2, vh2, 1);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:332:1: error: inlining failed in call to always_inline ‘__m256 _mm256_shuffle_ps(__m256, __m256, int)’: target specific option mismatch
 _mm256_shuffle_ps (__m256 __A, __m256 __B, const int __mask)
 ^
src/func/lookup.cpp:54:72: error: called from here
     const auto tmp0 = _mm256_shuffle_ps(v1, v2, _MM_SHUFFLE(2, 1, 3, 2));
                                                                        ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:332:1: error: inlining failed in call to always_inline ‘__m256 _mm256_shuffle_ps(__m256, __m256, int)’: target specific option mismatch
 _mm256_shuffle_ps (__m256 __A, __m256 __B, const int __mask)
 ^
src/func/lookup.cpp:55:72: error: called from here
     const auto tmp1 = _mm256_shuffle_ps(v0, v1, _MM_SHUFFLE(1, 0, 2, 1));
                                                                        ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:332:1: error: inlining failed in call to always_inline ‘__m256 _mm256_shuffle_ps(__m256, __m256, int)’: target specific option mismatch
 _mm256_shuffle_ps (__m256 __A, __m256 __B, const int __mask)
 ^
src/func/lookup.cpp:56:71: error: called from here
     const auto r = _mm256_shuffle_ps(v0, tmp0, _MM_SHUFFLE(2, 0, 3, 0));
                                                                       ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:332:1: error: inlining failed in call to always_inline ‘__m256 _mm256_shuffle_ps(__m256, __m256, int)’: target specific option mismatch
 _mm256_shuffle_ps (__m256 __A, __m256 __B, const int __mask)
 ^
src/func/lookup.cpp:57:73: error: called from here
     const auto g = _mm256_shuffle_ps(tmp1, tmp0, _MM_SHUFFLE(3, 1, 2, 0));
                                                                         ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:332:1: error: inlining failed in call to always_inline ‘__m256 _mm256_shuffle_ps(__m256, __m256, int)’: target specific option mismatch
 _mm256_shuffle_ps (__m256 __A, __m256 __B, const int __mask)
 ^
src/func/lookup.cpp:58:71: error: called from here
     const auto b = _mm256_shuffle_ps(tmp1, v2, _MM_SHUFFLE(3, 0, 3, 1));
                                                                       ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:454:1: error: inlining failed in call to always_inline ‘__m256i _mm256_cvtps_epi32(__m256)’: target specific option mismatch
 _mm256_cvtps_epi32 (__m256 __A)
 ^
src/func/lookup.cpp:59:63: error: called from here
     const auto ri = _mm256_cvtps_epi32(_mm256_mul_ps(r, coeff));
                                                               ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:306:1: error: inlining failed in call to always_inline ‘__m256 _mm256_mul_ps(__m256, __m256)’: target specific option mismatch
 _mm256_mul_ps (__m256 __A, __m256 __B)
 ^
src/func/lookup.cpp:59:39: error: called from here
     const auto ri = _mm256_cvtps_epi32(_mm256_mul_ps(r, coeff));
                                       ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:454:1: error: inlining failed in call to always_inline ‘__m256i _mm256_cvtps_epi32(__m256)’: target specific option mismatch
 _mm256_cvtps_epi32 (__m256 __A)
 ^
src/func/lookup.cpp:60:63: error: called from here
     const auto gi = _mm256_cvtps_epi32(_mm256_mul_ps(g, coeff));
                                                               ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:306:1: error: inlining failed in call to always_inline ‘__m256 _mm256_mul_ps(__m256, __m256)’: target specific option mismatch
 _mm256_mul_ps (__m256 __A, __m256 __B)
 ^
src/func/lookup.cpp:60:39: error: called from here
     const auto gi = _mm256_cvtps_epi32(_mm256_mul_ps(g, coeff));
                                       ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:454:1: error: inlining failed in call to always_inline ‘__m256i _mm256_cvtps_epi32(__m256)’: target specific option mismatch
 _mm256_cvtps_epi32 (__m256 __A)
 ^
src/func/lookup.cpp:61:63: error: called from here
     const auto bi = _mm256_cvtps_epi32(_mm256_mul_ps(b, coeff));
                                                               ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:306:1: error: inlining failed in call to always_inline ‘__m256 _mm256_mul_ps(__m256, __m256)’: target specific option mismatch
 _mm256_mul_ps (__m256 __A, __m256 __B)
 ^
src/func/lookup.cpp:61:39: error: called from here
     const auto bi = _mm256_cvtps_epi32(_mm256_mul_ps(b, coeff));
                                       ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:62:92: error: called from here
     const auto lr = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(lsb_ptr), ri, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:63:92: error: called from here
     const auto lg = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(lsb_ptr), gi, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:64:92: error: called from here
     const auto lb = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(lsb_ptr), bi, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:65:92: error: called from here
     const auto mr = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(msb_ptr), ri, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:66:92: error: called from here
     const auto mg = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(msb_ptr), gi, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:67:92: error: called from here
     const auto mb = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(msb_ptr), bi, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:682:1: error: inlining failed in call to always_inline ‘__m256i _mm256_slli_epi32(__m256i, int)’: target specific option mismatch
 _mm256_slli_epi32 (__m256i __A, int __B)
 ^
src/func/lookup.cpp:68:53: error: called from here
     const auto shifted_lg = _mm256_slli_epi32(lg, 10);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:682:1: error: inlining failed in call to always_inline ‘__m256i _mm256_slli_epi32(__m256i, int)’: target specific option mismatch
 _mm256_slli_epi32 (__m256i __A, int __B)
 ^
src/func/lookup.cpp:69:53: error: called from here
     const auto shifted_lb = _mm256_slli_epi32(lb, 20);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:682:1: error: inlining failed in call to always_inline ‘__m256i _mm256_slli_epi32(__m256i, int)’: target specific option mismatch
 _mm256_slli_epi32 (__m256i __A, int __B)
 ^
src/func/lookup.cpp:71:53: error: called from here
     const auto shifted_mg = _mm256_slli_epi32(mg, 10);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:682:1: error: inlining failed in call to always_inline ‘__m256i _mm256_slli_epi32(__m256i, int)’: target specific option mismatch
 _mm256_slli_epi32 (__m256i __A, int __B)
 ^
src/func/lookup.cpp:72:53: error: called from here
     const auto shifted_mb = _mm256_slli_epi32(mb, 20);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:899:1: error: inlining failed in call to always_inline ‘__m256i _mm256_unpacklo_epi32(__m256i, __m256i)’: target specific option mismatch
 _mm256_unpacklo_epi32 (__m256i __A, __m256i __B)
 ^
src/func/lookup.cpp:74:48: error: called from here
     const auto lo0 = _mm256_unpacklo_epi32(l, m);
                                                ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:871:1: error: inlining failed in call to always_inline ‘__m256i _mm256_unpackhi_epi32(__m256i, __m256i)’: target specific option mismatch
 _mm256_unpackhi_epi32 (__m256i __A, __m256i __B)
 ^
src/func/lookup.cpp:75:48: error: called from here
     const auto hi0 = _mm256_unpackhi_epi32(l, m);
                                                ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1081:1: error: inlining failed in call to always_inline ‘__m256i _mm256_permute2x128_si256(__m256i, __m256i, int)’: target specific option mismatch
 _mm256_permute2x128_si256 (__m256i __X, __m256i __Y, const int __M)
 ^
src/func/lookup.cpp:76:62: error: called from here
     const auto lo1 = _mm256_permute2x128_si256(lo0, hi0, 0x20);
                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1081:1: error: inlining failed in call to always_inline ‘__m256i _mm256_permute2x128_si256(__m256i, __m256i, int)’: target specific option mismatch
 _mm256_permute2x128_si256 (__m256i __X, __m256i __Y, const int __M)
 ^
src/func/lookup.cpp:77:62: error: called from here
     const auto hi1 = _mm256_permute2x128_si256(lo0, hi0, 0x31);
                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:900:1: error: inlining failed in call to always_inline ‘void _mm256_storeu_si256(__m256i*, __m256i)’: target specific option mismatch
 _mm256_storeu_si256 (__m256i *__P, __m256i __A)
 ^
src/func/lookup.cpp:78:78: error: called from here
     _mm256_storeu_si256(reinterpret_cast<__m256i*>(out_ptr + 2 * i + 0), lo1);
                                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:900:1: error: inlining failed in call to always_inline ‘void _mm256_storeu_si256(__m256i*, __m256i)’: target specific option mismatch
 _mm256_storeu_si256 (__m256i *__P, __m256i __A)
 ^
src/func/lookup.cpp:79:78: error: called from here
     _mm256_storeu_si256(reinterpret_cast<__m256i*>(out_ptr + 2 * i + 8), hi1);
                                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:1285:1: error: inlining failed in call to always_inline ‘__m256 _mm256_set1_ps(float)’: target specific option mismatch
 _mm256_set1_ps (float __A)
 ^
src/func/lookup.cpp:43:43: error: called from here
   const auto coeff = _mm256_set1_ps(255.0f);
                                           ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:1285:1: error: inlining failed in call to always_inline ‘__m256 _mm256_set1_ps(float)’: target specific option mismatch
 _mm256_set1_ps (float __A)
 ^
src/func/lookup.cpp:43:43: error: called from here
   const auto coeff = _mm256_set1_ps(255.0f);
                                           ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:1451:1: error: inlining failed in call to always_inline ‘__m256 _mm256_castps128_ps256(__m128)’: target specific option mismatch
 _mm256_castps128_ps256 (__m128 __A)
 ^
src/func/lookup.cpp:45:78: error: called from here
     const auto vl0 = _mm256_castps128_ps256(_mm_loadu_ps(in_ptr + 3 * i +  0));
                                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:1451:1: error: inlining failed in call to always_inline ‘__m256 _mm256_castps128_ps256(__m128)’: target specific option mismatch
 _mm256_castps128_ps256 (__m128 __A)
 ^
src/func/lookup.cpp:46:78: error: called from here
     const auto vl1 = _mm256_castps128_ps256(_mm_loadu_ps(in_ptr + 3 * i +  4));
                                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:1451:1: error: inlining failed in call to always_inline ‘__m256 _mm256_castps128_ps256(__m128)’: target specific option mismatch
 _mm256_castps128_ps256 (__m128 __A)
 ^
src/func/lookup.cpp:47:78: error: called from here
     const auto vl2 = _mm256_castps128_ps256(_mm_loadu_ps(in_ptr + 3 * i +  8));
                                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:734:1: error: inlining failed in call to always_inline ‘__m256 _mm256_insertf128_ps(__m256, __m128, int)’: target specific option mismatch
 _mm256_insertf128_ps (__m256 __X, __m128 __Y, const int __O)
 ^
src/func/lookup.cpp:51:53: error: called from here
     const auto v0 = _mm256_insertf128_ps(vl0, vh0, 1);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:734:1: error: inlining failed in call to always_inline ‘__m256 _mm256_insertf128_ps(__m256, __m128, int)’: target specific option mismatch
 _mm256_insertf128_ps (__m256 __X, __m128 __Y, const int __O)
 ^
src/func/lookup.cpp:52:53: error: called from here
     const auto v1 = _mm256_insertf128_ps(vl1, vh1, 1);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:734:1: error: inlining failed in call to always_inline ‘__m256 _mm256_insertf128_ps(__m256, __m128, int)’: target specific option mismatch
 _mm256_insertf128_ps (__m256 __X, __m128 __Y, const int __O)
 ^
src/func/lookup.cpp:53:53: error: called from here
     const auto v2 = _mm256_insertf128_ps(vl2, vh2, 1);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:332:1: error: inlining failed in call to always_inline ‘__m256 _mm256_shuffle_ps(__m256, __m256, int)’: target specific option mismatch
 _mm256_shuffle_ps (__m256 __A, __m256 __B, const int __mask)
 ^
src/func/lookup.cpp:54:72: error: called from here
     const auto tmp0 = _mm256_shuffle_ps(v1, v2, _MM_SHUFFLE(2, 1, 3, 2));
                                                                        ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:332:1: error: inlining failed in call to always_inline ‘__m256 _mm256_shuffle_ps(__m256, __m256, int)’: target specific option mismatch
 _mm256_shuffle_ps (__m256 __A, __m256 __B, const int __mask)
 ^
src/func/lookup.cpp:55:72: error: called from here
     const auto tmp1 = _mm256_shuffle_ps(v0, v1, _MM_SHUFFLE(1, 0, 2, 1));
                                                                        ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:332:1: error: inlining failed in call to always_inline ‘__m256 _mm256_shuffle_ps(__m256, __m256, int)’: target specific option mismatch
 _mm256_shuffle_ps (__m256 __A, __m256 __B, const int __mask)
 ^
src/func/lookup.cpp:56:71: error: called from here
     const auto r = _mm256_shuffle_ps(v0, tmp0, _MM_SHUFFLE(2, 0, 3, 0));
                                                                       ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:332:1: error: inlining failed in call to always_inline ‘__m256 _mm256_shuffle_ps(__m256, __m256, int)’: target specific option mismatch
 _mm256_shuffle_ps (__m256 __A, __m256 __B, const int __mask)
 ^
src/func/lookup.cpp:57:73: error: called from here
     const auto g = _mm256_shuffle_ps(tmp1, tmp0, _MM_SHUFFLE(3, 1, 2, 0));
                                                                         ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:332:1: error: inlining failed in call to always_inline ‘__m256 _mm256_shuffle_ps(__m256, __m256, int)’: target specific option mismatch
 _mm256_shuffle_ps (__m256 __A, __m256 __B, const int __mask)
 ^
src/func/lookup.cpp:58:71: error: called from here
     const auto b = _mm256_shuffle_ps(tmp1, v2, _MM_SHUFFLE(3, 0, 3, 1));
                                                                       ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:454:1: error: inlining failed in call to always_inline ‘__m256i _mm256_cvtps_epi32(__m256)’: target specific option mismatch
 _mm256_cvtps_epi32 (__m256 __A)
 ^
src/func/lookup.cpp:59:63: error: called from here
     const auto ri = _mm256_cvtps_epi32(_mm256_mul_ps(r, coeff));
                                                               ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:306:1: error: inlining failed in call to always_inline ‘__m256 _mm256_mul_ps(__m256, __m256)’: target specific option mismatch
 _mm256_mul_ps (__m256 __A, __m256 __B)
 ^
src/func/lookup.cpp:59:39: error: called from here
     const auto ri = _mm256_cvtps_epi32(_mm256_mul_ps(r, coeff));
                                       ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:454:1: error: inlining failed in call to always_inline ‘__m256i _mm256_cvtps_epi32(__m256)’: target specific option mismatch
 _mm256_cvtps_epi32 (__m256 __A)
 ^
src/func/lookup.cpp:60:63: error: called from here
     const auto gi = _mm256_cvtps_epi32(_mm256_mul_ps(g, coeff));
                                                               ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:306:1: error: inlining failed in call to always_inline ‘__m256 _mm256_mul_ps(__m256, __m256)’: target specific option mismatch
 _mm256_mul_ps (__m256 __A, __m256 __B)
 ^
src/func/lookup.cpp:60:39: error: called from here
     const auto gi = _mm256_cvtps_epi32(_mm256_mul_ps(g, coeff));
                                       ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:454:1: error: inlining failed in call to always_inline ‘__m256i _mm256_cvtps_epi32(__m256)’: target specific option mismatch
 _mm256_cvtps_epi32 (__m256 __A)
 ^
src/func/lookup.cpp:61:63: error: called from here
     const auto bi = _mm256_cvtps_epi32(_mm256_mul_ps(b, coeff));
                                                               ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:306:1: error: inlining failed in call to always_inline ‘__m256 _mm256_mul_ps(__m256, __m256)’: target specific option mismatch
 _mm256_mul_ps (__m256 __A, __m256 __B)
 ^
src/func/lookup.cpp:61:39: error: called from here
     const auto bi = _mm256_cvtps_epi32(_mm256_mul_ps(b, coeff));
                                       ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:62:92: error: called from here
     const auto lr = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(lsb_ptr), ri, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:63:92: error: called from here
     const auto lg = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(lsb_ptr), gi, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:64:92: error: called from here
     const auto lb = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(lsb_ptr), bi, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:65:92: error: called from here
     const auto mr = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(msb_ptr), ri, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:66:92: error: called from here
     const auto mg = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(msb_ptr), gi, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1593:1: error: inlining failed in call to always_inline ‘__m256i _mm256_i32gather_epi32(const int*, __m256i, int)’: target specific option mismatch
 _mm256_i32gather_epi32 (int const *__base, __m256i __index, const int __scale)
 ^
src/func/lookup.cpp:67:92: error: called from here
     const auto mb = _mm256_i32gather_epi32(reinterpret_cast<const int32_t*>(msb_ptr), bi, 4);
                                                                                            ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:682:1: error: inlining failed in call to always_inline ‘__m256i _mm256_slli_epi32(__m256i, int)’: target specific option mismatch
 _mm256_slli_epi32 (__m256i __A, int __B)
 ^
src/func/lookup.cpp:68:53: error: called from here
     const auto shifted_lg = _mm256_slli_epi32(lg, 10);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:682:1: error: inlining failed in call to always_inline ‘__m256i _mm256_slli_epi32(__m256i, int)’: target specific option mismatch
 _mm256_slli_epi32 (__m256i __A, int __B)
 ^
src/func/lookup.cpp:69:53: error: called from here
     const auto shifted_lb = _mm256_slli_epi32(lb, 20);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:682:1: error: inlining failed in call to always_inline ‘__m256i _mm256_slli_epi32(__m256i, int)’: target specific option mismatch
 _mm256_slli_epi32 (__m256i __A, int __B)
 ^
src/func/lookup.cpp:71:53: error: called from here
     const auto shifted_mg = _mm256_slli_epi32(mg, 10);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:682:1: error: inlining failed in call to always_inline ‘__m256i _mm256_slli_epi32(__m256i, int)’: target specific option mismatch
 _mm256_slli_epi32 (__m256i __A, int __B)
 ^
src/func/lookup.cpp:72:53: error: called from here
     const auto shifted_mb = _mm256_slli_epi32(mb, 20);
                                                     ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:899:1: error: inlining failed in call to always_inline ‘__m256i _mm256_unpacklo_epi32(__m256i, __m256i)’: target specific option mismatch
 _mm256_unpacklo_epi32 (__m256i __A, __m256i __B)
 ^
src/func/lookup.cpp:74:48: error: called from here
     const auto lo0 = _mm256_unpacklo_epi32(l, m);
                                                ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:871:1: error: inlining failed in call to always_inline ‘__m256i _mm256_unpackhi_epi32(__m256i, __m256i)’: target specific option mismatch
 _mm256_unpackhi_epi32 (__m256i __A, __m256i __B)
 ^
src/func/lookup.cpp:75:48: error: called from here
     const auto hi0 = _mm256_unpackhi_epi32(l, m);
                                                ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1081:1: error: inlining failed in call to always_inline ‘__m256i _mm256_permute2x128_si256(__m256i, __m256i, int)’: target specific option mismatch
 _mm256_permute2x128_si256 (__m256i __X, __m256i __Y, const int __M)
 ^
src/func/lookup.cpp:76:62: error: called from here
     const auto lo1 = _mm256_permute2x128_si256(lo0, hi0, 0x20);
                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:43:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avx2intrin.h:1081:1: error: inlining failed in call to always_inline ‘__m256i _mm256_permute2x128_si256(__m256i, __m256i, int)’: target specific option mismatch
 _mm256_permute2x128_si256 (__m256i __X, __m256i __Y, const int __M)
 ^
src/func/lookup.cpp:77:62: error: called from here
     const auto hi1 = _mm256_permute2x128_si256(lo0, hi0, 0x31);
                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:900:1: error: inlining failed in call to always_inline ‘void _mm256_storeu_si256(__m256i*, __m256i)’: target specific option mismatch
 _mm256_storeu_si256 (__m256i *__P, __m256i __A)
 ^
src/func/lookup.cpp:78:78: error: called from here
     _mm256_storeu_si256(reinterpret_cast<__m256i*>(out_ptr + 2 * i + 0), lo1);
                                                                              ^
In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include/immintrin.h:41:0,
                 from /usr/lib/gcc/x86_64-linux-gnu/5/include/x86intrin.h:46,
                 from src/func/lookup.cpp:20:
/usr/lib/gcc/x86_64-linux-gnu/5/include/avxintrin.h:900:1: error: inlining failed in call to always_inline ‘void _mm256_storeu_si256(__m256i*, __m256i)’: target specific option mismatch
 _mm256_storeu_si256 (__m256i *__P, __m256i __A)
 ^
src/func/lookup.cpp:79:78: error: called from here
     _mm256_storeu_si256(reinterpret_cast<__m256i*>(out_ptr + 2 * i + 8), hi1);
                                                                              ^
Makefile:238: recipe for target 'src/func/lookup.o' failed
make: *** [src/func/lookup.o] Error 1
```

--- a/dlk/python/dlk/templates/Makefile
+++ b/dlk/python/dlk/templates/Makefile
@@ -162,7 +162,7 @@ ar_x86:           NAME = x86
 
 ar_x86_avx:       AR = ar
 ar_x86_avx:       CXX = g++
-ar_x86_avx:       FLAGS += $(INCLUDES) -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -pthread -g -fopenmp
+ar_x86_avx:       FLAGS += $(INCLUDES) -O3 -std=c++14 -fPIC -fvisibility=hidden -DUSE_AVX -mavx2 -mfma -pthread -g -fopenmp
 ar_x86_avx:       LDFLAGS += -rcs
 ar_x86_avx:       NAME = x86_avx
 

